src/Compilers/CSharp/Portable/Binder/LookupResult.cs

```diff
@@ -191,7 +191,13 @@ internal static SingleLookupResult NotTypeOrNamespace(Symbol symbol, DiagnosticI
         internal static SingleLookupResult NotTypeOrNamespace(Symbol unwrappedSymbol, Symbol symbol, bool diagnose)
         {
             // TODO: determine correct diagnosis 
-            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
+            object sym = unwrappedSymbol;
+            // for CS0118 we dont want to display the type, but the name
+            if (unwrappedSymbol.Kind == SymbolKind.Parameter && diagnose)
```

Comment:
I think it's already sorted out, no? I was simply providing an explanation since you wrote this:

> Except for the initial comment, the rest of #7211 is very unclear to me.